workos/webhooks.py

```diff
@@ -34,19 +48,23 @@ def request_helper(self):
 
     DEFAULT_TOLERANCE = 180
 
-    def verify_event(self, payload, sig_header, secret, tolerance=DEFAULT_TOLERANCE):
-        if payload is None:
-            raise ValueError("Payload body is missing and is a required parameter")
-        if sig_header is None:
-            raise ValueError("Payload signature missing and is a required parameter")
-        if secret is None:
-            raise ValueError("Secret is missing and is a required parameter")
-
+    def verify_event(
+        self,
+        payload: WebhookPayload,
+        sig_header: str,
+        secret: str,
+        tolerance: Optional[int] = DEFAULT_TOLERANCE,
+    ) -> Webhook:
         Webhooks.verify_header(self, payload, sig_header, secret, tolerance)
-        event = json.loads(payload, object_pairs_hook=OrderedDict)
-        return event
-
-    def verify_header(self, event_body, event_signature, secret, tolerance=None):
+        return WebhookTypeAdapter.validate_json(payload)
+
+    def verify_header(
+        self,
+        event_body: WebhookPayload,
+        event_signature: str,
+        secret: str,
+        tolerance: Optional[int] = None,
+    ):
```

Comment:
Not a task for now, but does needing to wrap `verify_header()` in a `try/except` seem intuitive to you?

Reply:
Good question. It seems like the blessed path is to use `verify_event` (which actually shouldn't be called `verify_event`, it's really a deserialization). It makes sense for that to raise when the header can't be verified.

So what's the use case for using `verify_header` directly? Bypassing deserialization and parsing the webhook into a raw `dict`?

Reply:
I'm going to add a linear task to settle this.

Reply:
https://linear.app/workos/issue/DSYNC-1775/refactor-webhook-sdk-methods